src/ui/public/doc_table/components/table_row.js

```diff
@@ -92,18 +102,30 @@ module.directive('kbnTableRow', function ($compile) {
           openRowHtml
         ];
 
+        const mapping = indexPattern.fields.byName;
         if (indexPattern.timeFieldName) {
           newHtmls.push(cellTemplate({
             timefield: true,
-            formatted: _displayField(row, indexPattern.timeFieldName)
+            formatted: _displayField(row, indexPattern.timeFieldName),
+            filterable: mapping[indexPattern.timeFieldName].filterable,
+            column: indexPattern.timeFieldName
           }));
         }
 
         $scope.columns.forEach(function (column) {
+          const flattenedRow = indexPattern.flattenHit(row);
+          const isFilterable =
+            flattenedRow[column] === undefined
+            ? false
+            : !mapping[column]
+            ? false
+            : mapping[column].filterable;
```

Comment:
I had a few issues with this:

* I had to look up `_.get` to understand what each argument is for.
* I also had to hold a lot more state in my head to map input to output (like the double negative of "not undefined").
* Also, `get` returns the default value only for `undefined`, but this logic returns false if `mapping[column]` is falsy, so it's not quite a true replacement.

I think the reason I like the ternary is because it has the same pattern as a function with early exits, which makes it really easy for me to map input to output. Maybe the best route is to actually create a function that does just that?

```javascript
function isFilterable(row, mapping, column) {
  if (row[column] === undefined) {
    return false;
  }

  if (!mapping[column]) {
    return false;
  }

  return  mapping[column].filterable;
}
```

What do you think?